stm32-modules/include/heater-shaker/heater-shaker/gcodes.hpp

```diff
@@ -595,8 +597,29 @@ struct GetSystemInfo {
         if (written == write_to_limit) {
             return written;
         }
-        written = write_string_to_iterpair(written, write_to_limit,
-                                           serial_number.begin());
+        // If the serial number is unwritten, it will contain 0xFF which is
+        // an illegal character that will confuse the host side. Replace the
+        // first instance of it with a null terminator for safety.
+        constexpr uint8_t invalid_ascii_mask = 0x80;
```

Comment:
Let's not overcomplicate this - let's just look for the first exact match for `0xff`.